Core/Epaper/Src/epaper.c

```diff
@@ -0,0 +1,462 @@
+/* ==================================================================== */
+
+/*!
+  @brief   Pull epaper chip select pin low, allowing communication
+*/
+static void csOn()
+{
+    HAL_GPIO_WritePin(CS_EPD_GPIO_Port, CS_EPD_Pin, GPIO_PIN_RESET);
+}
+
+/*!
+  @brief   Pull epaper chip select pin high, inhibiting communication
+*/
+static void csOff()
+{
+    HAL_GPIO_WritePin(CS_EPD_GPIO_Port, CS_EPD_Pin, GPIO_PIN_SET);
+}
+
+/*!
+  @brief   Pull epaper data/command# pin low, allowing commands to be sent to the epaper
+*/
+static void setCommandMode()
+{
+    HAL_GPIO_WritePin(DC_GPIO_Port, DC_Pin, GPIO_PIN_RESET);
+}
+
+/*!
+  @brief   Pull epaper data/command# pin high, allowing data to be sent to the epaper
+*/
+static void setDataMode()
+{
+    HAL_GPIO_WritePin(DC_GPIO_Port, DC_Pin, GPIO_PIN_SET);
+}
+
+/*!
+  @brief   Pull epaper reset pin low, disabling the epaper
+*/
+static void reset()
+{
+	HAL_GPIO_WritePin(RST_GPIO_Port, RST_Pin, GPIO_PIN_RESET);
+	vTaskDelay(2 / portTICK_PERIOD_MS);
+}
+
+/*!
+  @brief   	Pull epaper reset pin high, enabling the epaper
+*/
+static void enable()
+{
+	HAL_GPIO_WritePin(RST_GPIO_Port, RST_Pin, GPIO_PIN_SET);
+	vTaskDelay(2 / portTICK_PERIOD_MS);
+}
+
+/*!
+  @brief   Wait for busy pin external interupt to be disasserted 
+*/
+static void waitBusyRelease()
+{
+    if(HAL_GPIO_ReadPin(BUSY_GPIO_Port, BUSY_Pin) == 1)
+    {
+        // Debug("e-Paper busy\r\n");
+        xSemaphoreTake(epdBusySemHandle, 0); // Guarantee epdBusySemHandle set to 0 
+        if (!(xSemaphoreTake(epdBusySemHandle, 5000) == pdTRUE))
+            {
+                Debug("Interrupt failed to occur during EPAP SPI transmit\n");
+            }
+        Debug("e-Paper busy release\r\n");
+    }
+}
+
+/*!
+  @brief   Send a command to the epaper
+  @param   command Register address of command 
+*/
+static void sendCommand(uint8_t command)
+{
+	// Set Epaper to command mode
+    setCommandMode();
+	// Open spi communication
+    csOn();
+
+	// Write command to spi 
+    HAL_SPI_Transmit_IT(&hspi2, (uint8_t *)&command, 1);
+	// Wait for spi interupt signaling data transfer complete - timeout 10ms
+    if (!(xSemaphoreTake(epdSpiSemHandle, 10) == pdTRUE))
+	{
+		Debug("Interrupt failed to occur during EPAP SPI transmit\n");
+	}
+
+	// Close spi communication
+    csOff();
+}
+
+/*!
+  @brief   Send a command followed by data to the epaper
+  @param   command 	Register address of command 
+  @param   data 	Data to be written to epaper 
+  @param   numBytes Size of data transfer excluding command byte 
+*/
+static void sendMessage(uint8_t command, uint8_t* data, uint16_t numBytes)
+{
+	    // Set Epaper to command mode
+		setCommandMode(); 
+		// Open spi communication	
+		csOn();
+
+		// Write command to spi 
+		HAL_SPI_Transmit_IT(&hspi2, (uint8_t *)&command, 1);
+		// Wait for spi interupt signaling data transfer complete - timeout 10ms
+		if (!(xSemaphoreTake(epdSpiSemHandle, 10) == pdTRUE))
```

Comment:
We should make the timeouts `#define TIMEOUT_10MS` to avoid magic numbers. I should probably also do that for the `bmbInterface.c` file